### PR TITLE
Support react@18

### PR DESCRIPTION
**Pull request by miafan23**
- Add react 18 to the allowed peer dependencies
- Bump @mapbox/react-submittable to allow react 18 as peer depemdency

### CODE REVIEW

File: src/components/accordion/__snapshots__/accordion.test.tsx.snap
Comment (tristen) [inline src/components/accordion/__snapshots__/accordion.test.tsx.snap:1]: What's up with these weird snapshot changes?

File: src/components/utils/use_previous.test.ts
Comment (tristen) [inline src/components/utils/use_previous.test.ts:1]: Good to know this has been consolidated into the latest version of `@testing-library/react`!

File: package.json [hunk 1]
```diff
@@ -30,12 +30,13 @@
...
     "prop-types": "^15.7.2",
-    "react": "^16.14.0",
-    "react-dom": "^16.14.0"
+    "react": "^16.14.0 || ^18.2.0",
+    "react-dom": "^16.14.0 || ^18.2.0"
   },
   "dependencies": {
     "@mapbox/mbx-assembly": "^1.5.0",
     "@mapbox/query-selector-contains-node": "^1.0.0",
+    "@mapbox/react-submittable": "^2.1.0-alpha.0",
```
Comment (tristen) [inline package.json:39]: What changes happened here besides the new package name? It's not great adding an `alpha` dependency if we don't need to.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mapbox/mr-ui",
-  "version": "2.9.0",
+  "version": "3.0.0",
```
Comment (tristen) [inline package.json:3]: Why is this a major version bump if it still supports `16`?

**Review by tristen — CHANGES_REQUESTED** [review]
@miafan23 left a couple of questions! Could you fill out the description of this PR for our future selves? It would be good to know about the necessity of this change, the Jira ticket this is references and the change to `react-submittable`.

File: src/components/accordion/__snapshots__/accordion.test.tsx.snap
Comment (miafan23) [reply src/components/accordion/__snapshots__/accordion.test.tsx.snap:1]: Radix changed the id for React 18 for better support of SSR
https://github.com/radix-ui/primitives/blob/main/packages/react/id/src/id.tsx#L10
https://www.radix-ui.com/primitives/docs/guides/server-side-rendering

File: package.json (hunk 1 above)
Comment (miafan23) [reply package.json:39]: Oh, I am just waiting for https://github.com/mapbox/react-submittable/pull/9 to be merged and released

**Comment by miafan23** [comment]
>the Jira ticket this is references

@tristen  This is a public repo, I am not sure if we should refer jira ticket here 🤔

File: src/components/tabs/tabs.test.tsx [hunk 3]
```diff
@@ -21,14 +21,18 @@ describe('Tabs', () => {
         tab1.focus();
         tab1.click();
 
-        expect(screen.getByText('Tab 1 content')).toBeInTheDocument();
+        await waitFor(() => {
+            expect(screen.getByText('Tab 1 content')).toBeInTheDocument();
+        })
```
Comment (miafan23) [inline src/components/tabs/tabs.test.tsx:26]: These tests sometime failed

**Review by tristen — APPROVED** [review]

**Pull request merged**